### PR TITLE
Prepare for gcc-10

### PR DESCRIPTION
**Pull request by maierlars**
### Scope & Purpose

Fix compilation issues with the new gcc-10 compiler.

- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by maierlars** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/9968/

File: tests/Aql/MultiDependencySingleRowFetcherTest.cpp [hunk 1]
```diff
@@ -710,7 +714,7 @@ TEST_P(MultiDependencySingleRowFetcherTest,
   // This has to be done by the InputRange hold externally.
   // It has seperate tests
   std::vector<std::deque<SharedAqlItemBlockPtr>> data;
-  for (size_t i = 0; i < numberDependencies(); ++i) {
+  for (int i = 0; i < numberDependencies(); ++i) {
```
Comment (markuspf) [inline tests/Aql/MultiDependencySingleRowFetcherTest.cpp:717]: is this intended? is `numberDependencies()` an `int` that needs a sign?

Comment (maierlars) [reply tests/Aql/MultiDependencySingleRowFetcherTest.cpp:717]: no, but `i` needs to be signed ... this is just a quick and dirty fix. please fix

**Comment by maierlars** [comment]
image

**Pull request merged**